generator/src/main/java/com/google/archivepatcher/generator/FileByFileDeltaGenerator.java

```diff
@@ -118,4 +127,12 @@ private PreDiffPlan generatePreDiffPlan(
   protected DeltaGenerator getDeltaGenerator() {
     return new BsDiffDeltaGenerator();
   }
+
+  private static <T> List<T> getImmutableCopy(List<T> input) {
+    if (input != null) {
+      return Collections.unmodifiableList(input);
```

Comment:
This isn't creating an immutable copy - it's just making a wrapper around the input, which itself has an unspecified behavior with regards to mutation. If you pass in a mutable list to this method and subsequently modify the list that you passed in, the "copy" will have the changes too.